core/src/internal/douglas_peucker.cpp

```
@@ -0,0 +1,110 @@
+        const float retract = -interior_sign * d;
+        if (retract > max_retract) max_retract = retract;
+    }
+
+    if (max_abs_dev <= eps && max_retract <= retract_eps) return;  // drop the run
+
+    keep[split] = 1;
+    dp_reduce(pts, lo, split, eps, retract_eps, interior_sign, keep);
+    dp_reduce(pts, split, hi, eps, retract_eps, interior_sign, keep);
+}
+
+void dp_curve_reduction(const std::vector<std::vector<Point>> &chains,
```

Cppcheck (2.20.0)

[style] The function 'dp_curve_reduction' is never used.

(unusedFunction)